src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      callback(number, this);
+
+      return this;
+    },
+    add(number, calculator) {
+      calculator.result += number;
+    },
+    subtract(number, calculator) {
+      calculator.result -= number;
```

Comment:
The `subtract` method should not take the calculator as a parameter. It should use the `this` context to modify the `result` property directly.